### PR TITLE
feat: Use environment entity list instead of schema file for import mappings form

### PR DESCRIPTION
**Pull request by danroche10**

### CODE REVIEW

File: Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/Presenters/ImportMappingsFormPresenter.cs [hunk 1]
```diff
@@ -23,16 +30,19 @@ public ImportMappingsFormPresenter(IImportMappingsFormView view)
...
+            if (OrganizationService == null)
             {
-                ViewHelpers.ShowMessage("Please specify a schema file with atleast one entity defined.", "No entities available", MessageBoxButtons.OK, MessageBoxIcon.Error);
                 view.Close();
+                ViewHelpers.ShowMessage("Please make sure you are connected to an organisation", "No connection made", MessageBoxButtons.OK, MessageBoxIcon.Information);
                 return;
             }
-
-            view.EntityList = view.SchemaConfiguration.Entities.Select(x => x.Name).OrderBy(n => n);
+            if (new List<string>(view.EntityListDataSource).Count == 0)
```
Comment (tdashworth) [inline Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/Presenters/ImportMappingsFormPresenter.cs:39]: What is this condition for?

File: Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/UserControls/ImportPage.cs [hunk 2]
```diff
@@ -118,6 +114,24 @@ string IImportPageView.AskForFilePathToSave(string existingFileName)
...
+        public void SetOrganizationService(IOrganizationService organizationService)
+        {
+            importLookupMappingsFormPresenter.OrganizationService = organizationService;
+        }
+
+        [ExcludeFromCodeCoverage]
+        public void SetViewHelpers(IViewHelpers viewHelpers)
+        {
+            importLookupMappingsFormPresenter.ViewHelpers = viewHelpers;
+        }
```
Comment (tdashworth) [inline Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/UserControls/ImportPage.cs:133]: What are your thoughts on merging these into `SetServices(...)`?

File: Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/UserControls/ExportPage.cs [hunk 3]
```diff
@@ -176,7 +148,32 @@ string IExportPageView.AskForFilePathToSave(string existingFileName)
...
+        {
+            exportLookupMappingsFormPresenter.ExceptionService = exceptionService;
+        }
+
+        [ExcludeFromCodeCoverage]
+        public void SetViewHelpers(IViewHelpers viewHelpers)
+        {
+            exportLookupMappingsFormPresenter.ViewHelpers = viewHelpers;
+            exportFilterFormPresenter.ViewHelpers = viewHelpers;
+        }
```
Comment (tdashworth) [inline Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/UserControls/ExportPage.cs:175]: Consider merging like `ImportPage`?

**Review by tdashworth — CHANGES_REQUESTED** [review]

**Comment by tdashworth** [comment]
Thanks @danroche10! Just a few minor changes/suggestions. Let me know what you think.

File: Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/UserControls/ImportPage.cs (hunk 2 above)
Comment (danroche10) [reply Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/UserControls/ImportPage.cs:133]: @tdashworth - Yeah makes sense. And include SetViewHelpers in this?

File: Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/Presenters/ImportMappingsFormPresenter.cs (hunk 1 above)
Comment (danroche10) [reply Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/Presenters/ImportMappingsFormPresenter.cs:39]: It's to check whether EntityListDataSource has already been set. We only retrieve entities if it hasn't already been set

Comment (tdashworth) [reply Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/Presenters/ImportMappingsFormPresenter.cs:39]: Ah yes! I was confused about the `new List()`. Would this not work
```suggestion
            if (!view.EntityListDataSource.Any())
```

**Comment by danroche10** [comment]
> Thanks @danroche10! Just a few minor changes/suggestions. Let me know what you think.

@tdashworth - implemented requested changes!

**Pull request merged**